ros_ws/src/get_user_request.py

```diff
@@ -0,0 +1,141 @@
+import os
+import langchain
+
+from langchain.llms import OpenAI
+from langchain import PromptTemplate
+from langchain import schema
+from langchain.prompts import (ChatPromptTemplate,PromptTemplate,SystemMessagePromptTemplate, HumanMessagePromptTemplate)
+from langchain.chat_models import ChatOpenAI
+from langchain.chains import LLMChain
+from langchain.schema import HumanMessage, SystemMessage, AIMessage
+api_key=os.environ["OPENAI_API_KEY"]
+model_to_use="gpt-4-0613"
+#model_to_use="text-davinci-003"
+
+
+def Refine(request):
+    chat = ChatOpenAI(model_name=model_to_use)
+
+    sys_msg=SystemMessage(content="""You are a home assistant robot that can retrieve objects for your user. 
+        You are looking for a single item to search for.  
+        You should ask clarifying questions of your user, only if needed, until you have a 
+        sufficient description of a specific item to search for and retrieve.
+        If the item seems vague or general or the user isn't sure, make a suggestion.
+        If the user requests "something" or "anything", you shouold make a suggestion that is consistent
+        with the rest of the request.  For example, if the user requests please bring me anything cold,
+        you might suggest "would you like a glass of water?" """)
+   
+    unknown_request=True
+    context=[sys_msg]
+    attempts_to_understand=0
+    user_msg="User: " + request
+
+    print (user_msg)
```

Comment:
Functions shouldn't print except in debugging or if that's the desired output. ROS2 logging modules is a different story.